sdk/cosmos/microsoft-azure-cosmos-benchmark/src/main/java/com/azure/data/cosmos/benchmark/ReadMyWriteWorkflow.java

```diff
@@ -124,8 +132,14 @@ protected void performWorkload(BaseSubscriber<Document> baseSubscriber, long i)
         }
 
         concurrencyControlSemaphore.acquire();
+        logger.debug("concurrencyControlSemaphore: {}", concurrencyControlSemaphore);
 
-        obs.subscribeOn(Schedulers.parallel()).subscribe(baseSubscriber);
+        try {
+            obs.subscribeOn(Schedulers.parallel()).subscribe(baseSubscriber);
+        } catch (Throwable error) {
+            concurrencyControlSemaphore.release();
+            logger.error("subscription failed due to ", error);
```

Comment:
try/catch for subscribe should not be needed. please remove, unless you have seen a failure from here in that case please share the stacktrace here.

Reply:
thanks for finding this. it is now corrected.